build/kubernetes/run-local.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+NAMESPACE=${1-infinispan}
+
+echo "Using kubeconfig: ${KUBECONFIG}"
+echo "Running for cluster: $(kubectl config current-context)"
+
+kubectl apply -f deploy/crds/infinispan.org_infinispans_crd.yaml
+kubectl apply -f deploy/crds/infinispan.org_caches_crd.yaml
+kubectl wait --for condition=established crd infinispans.infinispan.org --timeout=60s
+WATCH_NAMESPACE="${NAMESPACE}" ./build/_output/bin/infinispan-operator -kubeconfig "${KUBECONFIG}"
```

Comment:
I'm not using minikube/k8s, but seems to be no specific for run/run-local between plain k8s and minikube.
It would be nice to split and don't duplicate k8s/minikube commons, like run, run-local and clean that's not specific for minikube and minukube start/stop/delete scipts.
@rigazilla wdyt about it?

Reply:
Also would be fine to have common scripts for both ocp/k8s as difference between `oc` and `kubectl` utility is very limited. Just for our scripts it's maybe `oc new-project` that's equivalent `kubectl create ns` nothing more. Otherwise all commands are the same...

Reply:
Are you suggesting that `oc` commands not be used for `run` or `run-local`, and `kubectl` be used instead in a common script?
There is also not an equivalent command for `oc login` that I know of, but the docs could instruct the user to run that separately and then populate the `KUBECONFIG` environment variable.

Reply:
I think it would definitely be doable to extract the unique commands to the docs, and put everything else inside common `run` and `run-local` scripts which use `kubectl`.

Reply:
First, let's remove all kubernetes/minikube things from this PR and arrange this in separate one. We can create a new issue for this. I can do this, but can't test as using OCP in AWS only and don't have alternative one.
@timothymathison you are welcome to do this.

Second, it would be nice to separate minikube things from plain k8s. For example, keep just minikube related scripts in minikube folder and make targets named as minikube just for them. Place kubernetes related scripts such as test, run-local, etc. in kubernetes folder and change make targets for kubernetes only not related to minikube.
As maximum target it would be nice to have common ocp/k8s scripts for run-local, run, etc. which will run with just parameter depends of the make target which will define where is this script running, on ocp or k8s independently deployment platform for k8s, such as minikube, kind or plain k8s on baremetal.
Hope, I explain correctly, sorry for possible grammatical mistakes, I'm not native :smile: